File: Number of divisors/Number of divisors.py
class Solution:
    def count_divisors(self, N):
        count = 0
        sqrt_N = int(N ** 0.5)
        for i in range(1, sqrt_N + 1):
            if N % i == 0:
                if i % 3 == 0:
                    count += 1
                divisor = N // i
                if divisor != i and divisor % 3 == 0:
                    count += 1
        return count

File: Number of divisors/Number of divisors.cpp
class Solution
{
	public:
		int count_divisors(int n)
		{
    int c=0;
    
    for(int a=1; a*a<=n; a++){
        if(n%a==0){
            if(a==n/a){
                if(a%3==0)
                    c++;
            }else{
                if(a%3==0){
                    c++;
                }
                if((n/a)%3==0)
                    c++;
            }
        }
    }
    return c;
		}
};